src/main/java/land/oras/OCI.java

```diff
@@ -421,4 +419,98 @@ public Manifest attachArtifact(T ref, ArtifactType artifactType, Annotations ann
+        }
+        visited.add(digest);
+
+        // Read the manifest or index
+        Path blobPath = layout.getBlobPath(digest);
+        String content = Files.readString(blobPath);
+        Map<String, Object> node = JsonUtils.fromJson(content, Map.class);
+
+        // Handle index (has "manifests")
+        if (node.containsKey("manifests")) {
+            List<Map<String, Object>> manifests = (List<Map<String, Object>>) node.get("manifests");
+            for (Map<String, Object> manifest : manifests) {
+                String manifestDigest = (String) manifest.get("digest");
+                if (manifestDigest != null) {
+                    digests.add(manifestDigest);
+                    digests.addAll(collectReferencedDigests(layout, manifestDigest, visited));
+                }
+            }
+        } else {
+            // Handle manifest
+            // Collect config digest
+            if (node.containsKey("config")) {
+                Map<String, Object> config = (Map<String, Object>) node.get("config");
+                String configDigest = (String) config.get("digest");
+                if (configDigest != null) {
+                    digests.add(configDigest);
+                }
+            }
+            // Collect layer digests
+            if (node.containsKey("layers")) {
+                List<Map<String, Object>> layers = (List<Map<String, Object>>) node.get("layers");
+                for (Map<String, Object> layer : layers) {
+                    String layerDigest = (String) layer.get("digest");
+                    if (layerDigest != null) {
+                        digests.add(layerDigest);
+                    }
+                }
+            }
+            // Handle subject (for reference artifacts)
+            if (node.containsKey("subject")) {
+                Map<String, Object> subject = (Map<String, Object>) node.get("subject");
+                String subjectDigest = (String) subject.get("digest");
+                if (subjectDigest != null) {
+                    digests.add(subjectDigest);
+                    digests.addAll(collectReferencedDigests(layout, subjectDigest, visited));
+                }
+            }
+        }
+
+        return digests;
+    }
+
+    /**
+     * Get the blob path for a given digest.
+     *
+     * @param digest the digest (e.g., sha256:abc123)
+     * @return the file path to the blob
+     */
+    public Path getBlobPath(String digest) {
+        String[] parts = digest.split(":");
+        if (parts.length != 2) {
+            throw new IllegalArgumentException("Invalid digest format: " + digest);
+        }
+        String algorithm = parts[0];
+        String hash = parts[1];
+        if (this instanceof OCILayout) {
+            OCILayout layout = (OCILayout) this;
+            return layout.getPath().resolve("blobs").resolve(algorithm).resolve(hash);
+        }
+        throw new UnsupportedOperationException("getBlobPath not supported in this context");
```

Comment:
OCI is a sealed class with only Registry and OCI layout. So why adding this method on parent if not supported by Registry?